app/controllers/readings_controller.rb

```diff
@@ -14,30 +16,25 @@ def index
     end
   end
 
-  # GET /readings/1 or /readings/1.json
-  def show
-  end
-
   # GET /readings/new
   def new
     @reading = Reading.new
+    authorize @reading, :new? 
     @user_vitals = current_user.vitals
   end
 
   # GET /readings/1/edit
   def edit
-    @reading = Reading.find(params[:id])
     @user_vitals = current_user.vitals
   end
 
   # POST /readings or /readings.json
   def create
-    # Assuming `vital_id` is passed as part of the form submission
     vital = current_user.vitals.find(params[:reading][:vital_id])
-  
-    # Initialize the reading with the user_id set to the current_user's id
     @reading = Reading.new(reading_params.merge(user_id: current_user.id, vital_id: vital.id))
   
+    authorize @reading
+
```

Comment:
refactor since you already have `before_action :authorize_reading, only: %i[show edit update destroy]` on line 4

Reply:
So, as I understand it, if I move this to the ``before_action :authorize_reading`` action, @reading is not yet set when the action starts since it depends on reading_params and other dynamic data that's not available until partway through the action. So if I call it before setting the @reading value (using params) it'll give me a nil value error. Maybe that's an issue with how I'm structuring my logic? It seems only to work if I place the ``authorize @reading`` after passing the dynamic param values into the @reading variable.

Reply:
I see, I think you are right. There may be a way to change how @reading is assigned here so it works. I'll look at this tomorrow.

Reply:
Ok thanks! Yeah no pressure to reply to me tonight 😅

Reply:
Actually, changing line 4 to do `before_action { authorize(@reading || Reading) }` would fix this `nil` value error, making the code a lot more DRY.

Basically, we are not really looking for an authorization rule on the specific instance of `@reading` here, so we can call `authorize` on the Class itself.

Reply:
Thank you! I was able to refactor using that. It also automatically gets rid of the authorize_reading method you mentioned not being necessary earlier. 😃